src/Google.Pubsub.V1/SimpleSubscription.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using static Google.Pubsub.V1.PublisherClient;
+using static Google.Pubsub.V1.SubscriberClient;
+
+
+namespace Google.Pubsub.V1
+{
+    public class SimpleSubscription
+    {
+        const int MaxMessages = 10;
+
+        public string ProjectId => Parent.ProjectId;
+        public string SubscriptionId { get; }
+        public string TopicId { get; }
+        public string TopicResourceName { get; }
+        public string ResourceName { get; }
+        public SimplePubsub Parent { get; }
+
+        public SimpleSubscription(SimplePubsub parent, string subscriptionId, string topicId)
+        {
+            Parent = parent;
+            SubscriptionId = subscriptionId;
+            TopicId = topicId;
+            ResourceName = GetSubscriptionName(parent.ProjectId, subscriptionId);
+            TopicResourceName = GetTopicName(parent.ProjectId, topicId);
+        }
+
+        // Options for completion:
+        // - CancellationToken
+        // - Return an IDisposable, and disposing of it stops (as per Rx)
+        // Other option: create another class which exposes an event. Or use IObservable.
+        // There are plenty of options for the async case, too...
+        // If we continue with the cancellation token, should we throw if the cancellation token is cancelled, or just return?
+
+        // Need to work out error handling - should we stop polling if the action throws an exception? (We may have acked other messages...)
+
+        public void Poll(Action<PubsubMessage> action, CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var messages = PullAndAcknowledge();
+                foreach (var message in messages)
+                {
+                    action(message);
+                }
+            }
+        }
+
+        public async Task PollAsync(Action<PubsubMessage> action, CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var messages = await PullAndAcknowledgeAsync(cancellationToken);
+                foreach (var message in messages)
+                {
+                    action(message);
+                }
+            }
+        }
+
+        public IList<PubsubMessage> PullAndAcknowledge()
+        {
+            var response = Parent.Subscriber.Pull(ResourceName, false, MaxMessages);
```

Comment:
In general, it might be better to prefer not to use "var" when it's not obvious what the type is, to make the code easier to read/understand.
What do you think?